tests/manage/nfs_feature/test_nfs_feature_enable_for_ODF_clusters.py

```diff
@@ -0,0 +1,481 @@
+            resource_count=self.num_of_cephfsplugin_pods,
+            condition=constants.STATUS_RUNNING,
+            selector="app=csi-nfsplugin",
+            dont_allow_other_resources=True,
+            timeout=60,
+        )
+
+        assert self.pod_obj.wait_for_resource(
+            resource_count=self.num_of_cephfsplugin_provisioner_pods,
+            condition=constants.STATUS_RUNNING,
+            selector="app=csi-nfsplugin-provisioner",
+            dont_allow_other_resources=True,
+            timeout=60,
+        )
+
+        # Fetch the nfs-ganesha pod name
+        pod_objs = pod.get_all_pods(
+            namespace=self.namespace, selector=["rook-ceph-nfs"]
+        )
+        log.info(f"pod objects---{pod_objs[0]}")
+        nfs_ganesha_pod_name = pod_objs[0].name
+        log.info(f"pod objects---{nfs_ganesha_pod_name}")
+
+        return nfs_ganesha_pod_name
+
+    def nfs_disable(self, nfs_ganesha_pod_name):
+        """
+        Args:
+            nfs_ganesha_pod_name: name of nfs-ganesha pod returned from nfs_enable()
+
+        Steps:
+        1. oc patch -n openshift-storage storageclusters.ocs.openshift.io ocs-storagecluster
+        --patch '{"spec": {"nfs":{"enable": false}}}' --type merge
+        2. oc patch cm rook-ceph-operator-config -n openshift-storage -p $'data:\n "ROOK_CSI_ENABLE_NFS":  "false"'
+        3. manually delete CephNFS, ocs nfs Service and the nfs StorageClass
+        4. Wait untill nfs-ganesha pod deleted
+        """
+
+        assert self.storage_cluster_obj.patch(
+            resource_name="ocs-storagecluster",
+            params=self.nfs_spec_disable,
+            format_type="merge",
+        ), "storagecluster.ocs.openshift.io/ocs-storagecluster not patched"
+
+        # Enable ROOK_CSI_ENABLE_NFS via patch request
+        assert self.config_map_obj.patch(
+            resource_name="rook-ceph-operator-config",
+            params=self.rook_csi_config_disable,
+            format_type="merge",
+        ), "configmap/rook-ceph-operator-config not patched"
+
+        # Delete the nfs StorageClass
+        self.sc.delete()
+
+        # Delete CephNFS
+        cmd_delete_cephnfs = "delete CephNFS ocs-storagecluster-cephnfs"
+        self.storage_cluster_obj.exec_oc_cmd(cmd_delete_cephnfs)
+
+        # Wait untill nfs-ganesha pod deleted
+        self.pod_obj.wait_for_delete(resource_name=nfs_ganesha_pod_name)
+
+    def test_nfs_not_enabled_by_default(
+        self,
+    ):
+        """
+        This test is to validate nfs feature is not enabled by default for  ODF(4.11) clusters
+
+        Steps:
+        1:- Check cephnfs resources not available by default
+
+        """
+        # Checks cephnfs resources not available by default
+        cephnfs_resource = self.storage_cluster_obj.exec_oc_cmd("get cephnfs")
+        if cephnfs_resource is None:
+            log.info(f"No resources found in openshift-storage namespace.")
+
+    def test_nfs_feature_enable(
```

Comment:
Are these tests intended to be run in parallel?

I notice that each test enables NFS, does some checking, then disables NFS. If the tests are intended to run in parallel, this is good. If they are sequential, it would be faster to set up NFS once, do all the tests, then disable NFS at the end.

Reply:
Addressed in latest commit.